### ISSUE TITLE
Cuda implementation of AKAZE

### ISSUE THREAD

**Issue by solonovamax** [alicevision/AliceVision#1757]
It would be nice if a cuda implementation of AKAZE could be added.

According to the comment on the `src/aliceVision/feature/akaze/AKAZE.hpp`, the implementation is taken from [pablofdezalc/akaze](https://github.com/pablofdezalc/akaze).

going to that page, they list the following under an "other implementations" header:
> You can try the CUDA A-KAZE implementation from Niklas Bergström: [nbergst/akaze](https://github.com/nbergst/akaze)
>
> There is also a very fast CPU A-KAZE version from Hideaki Suzuki [h2suzuki/fast_akaze](https://github.com/h2suzuki/fast_akaze)

Since those are all licensed under a BSD-3-Clause license, would it perhaps be possible to get the cuda version (and possible the "fast cpu" version) added to alicevision?

**Comment by natowi** [comment]
I agree, it would be great to have more cuda acceleration. You are welcome to contribute the implementation :)
If you are interested and want to get started, feel free to ask, the core devs are happy to give you some pointers.

**Comment by solonovamax** [comment]
> I agree, it would be great to have more cuda acceleration. You are welcome to contribute the implementation :)
> If you are interested and want to get started, feel free to ask, the core devs are happy to give you some pointers.

I'm not a c++ dev myself, which is why I didn't offer to do it. I don't think that c++ with archaic cuda code optimized for performance is the *best* place to get started with c++. but if I ever get around to learning c++, and this issue is still open, I *may* come back to it later

**Comment by natowi** [comment]
You are also welcome to help out with https://github.com/alicevision/Meshroom which is mostly Python/QT based. And if you´d like to tinker around a little, there is https://github.com/alicevision/MeshroomResearch. A plugin system is wip https://github.com/alicevision/Meshroom/pull/2407 to lower the contribution barrier to add new functionality.

**Comment by solonovamax** [comment]
not stale
